src/sql/src/query_model/dot.rs

```diff
@@ -47,7 +47,17 @@ impl DotGenerator {
     }
 
     /// Generates a graphviz graph for the given model, labeled with `label`.
-    pub fn generate(mut self, model: &Model, label: &str) -> Result<String, anyhow::Error> {
+    pub fn generate(self, model: &Model, label: &str) -> Result<String, anyhow::Error> {
+        self.generate_subgraph(model, model.top_box, label)
+    }
+
+    /// Generates a graphviz graph for the given subgraph of the model, labeled with `label`.
+    pub fn generate_subgraph(
+        mut self,
```

Comment:
nit: no need to have `mut self` here, as `generate` calls `generate_subgraph` but does not have `mut self`.

Reply:
The compiler says it is needed:

```
error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> src/sql/src/query_model/dot.rs:77:21  
   |               
56 |         self,                                                                                      
   |         ---- help: consider changing this to be mutable: `mut self` 
...                              
77 |                     self.new_line(&format!("subgraph cluster{} {{", box_id));                      
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
                                                                                                        
error[E0596]: cannot borrow `self` as mutable, as it is not declared as mutable
  --> src/sql/src/query_model/dot.rs:78:21  
   |               
56 |         self,                                                                                      
   |         ---- help: consider changing this to be mutable: `mut self` 
...                                                 
78 |                     self.inc();                                                                    
   |                     ^^^^^^^^^^ cannot borrow as mutable 
```

Reply:
I don't understand how the borrow checker does not complain about the 

```rust
self.generate_subgraph(model, model.top_box, label)
```

call in 

```rust
pub fn generate(self, model: &Model, label: &str) -> Result<String, anyhow::Error> { ... }
```

in that case...

Reply:
because `self` is passed to `generate` by value and that value is forwarded to `generate_subgraph`. Within `generate_subgraph`, `self` is just yet another variable that needs to be declared as mutable in order for `mut` methods to be invoked,